src/services/ui/src/features/package-actions/lib/correctStatusSwitch.ts

```diff
@@ -0,0 +1,45 @@
+import { type getItem } from "@/api";
+import { Action, SEATOOL_STATUS } from "shared-types";
+
+export const getStatusFor = (a: Action) => {
```

Comment:
this isn't very self explanatory and could be better structured and re-named to indicate its intent, and how the data pairings relate to each other.

you've also left some duplicate names in place from copying and pasting

Reply:
I renamed it. If you think more is needed, let me know. Basically the purpose here is to be able to effectively choose the correct status in which the data poller stops and says it is now synced

Reply:
Talked about migrating the check functions here to the same pattern that Action checks follow, with a PackageCheck param instead of the data param. Storing these with each action's index module sounds like a good idea to consolidate the scattered parts